### ISSUE TITLE
full stack framework？？？

### ISSUE THREAD

**Issue by fengahan** [amphp/http-server#318]
There is no intention of using it to open up a full stack framework or there is currently no implementation of this

**Comment by kelunik** [comment]
There's currently no implementation I'm aware of. Which features would you like to see?

**Comment by fengahan** [comment]
I would like to see a full stack framework like Laravel or YII implemented

**Comment by kelunik** [comment]
Again, please provide a list of features you're missing specifically.

**Comment by kelunik** [comment]
@fengahan Such behavior isn't useful at all, because there's nothing to base improvements on. Many components like sessions / routing etc. already exist, so a list of missing features would really be helpful.

**Comment by enumag** [comment]
@kelunik Let me try to help instead. I think what's missing isn't necessarily a framework as much as some guidelines / best practices / examples how to actually write a real-world application based on Amphp (and why bother at all).

There is http-server with sessions and such but how to implement a form? Sending, validation, rendering... Can I use any existing library or not?

Then any good app needs a DI container. Ofc you're free to use any with Amphp - but you have to create and configure all the services yourself, there is no helping hand. I'm not saying it's necessary, it's not that difficult to do yourself and you get some benefits from doing it too. But it's what people are used to and what they're expecting these days. It's not easy to actually build and amphp from scratch while also learning async so an example app would help quite a bit I think.

Then how to handle database operations. People are used to ORMs these days. Which ofc does have both benefits and downsides so a recommendation not to use one is valid. But you don't have an *option* to use one since none exists.

Then ofc you need to integrate a templating system such as Twig which again raises some questions about sync / async usage.

Then how do you implement a REST API or something? Can't really use any existing implementation since they're all based on some framework. Again doable but everything is "do it yourself".

Basically the ecosystem around Amphp is scarce and there aren't libraries and "how-do-i-do-X?" answers available on stackoverflow. And you can't just integrate and use any already existing framework's ecosystem. Maybe some parts with care but it's not straightforward.

Note: I'm personally not much interested in answers to these questions - my app is Amphp based with many SF libraries but it's GraphQL API so I don't need forms or templating and I managed to cut out Doctrine as well (actually I'm still using Doctrine in some places which don't need to be async + for migrations). I just think that's what the author of this issue might have been looking for.

**Comment by fengahan** [comment]
@kelunik Is the performance much lower than eventHttp
